src/plugins/action_trim_responses/src/trimResponses.ts

```diff
@@ -54,9 +54,19 @@ module.exports['trimResponses'] = async function (input: any, DBConnect: any) {
         latestResponse.value?.decision === null // Review responses
       )
         responsesToDelete.push(latestResponse.id)
-      else responsesToUpdate.push(latestResponse.id)
+      // Review responses to update calculation is seperate
+      else if (!reviewId) responsesToUpdate.push(latestResponse.id)
     })
 
+    // For level 1 reviewer:
+    // We want to update latest review responses if they are not null or duplicates
+    // If we update the latest review response for template element it should be
+    // review response that was changed in this review (not changed or null would be deleted in responsesToDelete)
+    if (reviewId && reviewLevel === 1)
+      responsesToUpdate = Object.values(groupResponses(responses, 'template_element_id')).map(
+        (responseArray) => responseArray[responseArray.length - 1].id
```

Comment:
From what I can tell, the `responsesToUpdate` array should be empty by here if it's a review (the above loop only pushes application responses to it). So why not just push on to the array here, then you go back to `const responsesToUpdate: number[] = []` up the top?
```
      responsesToUpdate.push(...Object.values(groupResponses(responses, 'template_element_id')).map(
        (responseArray) => responseArray[responseArray.length - 1].id
      ))
```

And you can chuck a `responsesToUpdate.length = 0` before just to be sure.

Sorry, that's super nit picky, I just try and avoid reassigning objects and arrays (always `const`), but its no real issue in this case.

Reply:
good point, come to think of it, this might be simpler ? c68fd1ddd34b0e97874e82e93df6e22533d3d2e3
I'll wait for your comment before mergin

Reply:
tests passed

Reply:
i just looked at that last commit and changed the name of variable: 315b1b9b262ec37fa2cda4ecc26ee2e798831a3f

Reply:
Yeah, LGTM :) Seems nicely logically broken up now.

I haven't checked, but did you confirm the tests still pass?

Reply:
yep, just re-ran again to be sure and did a couple of review cycles. cheers for review !